src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -842,7 +842,11 @@ private void finish(@Nonnull Result r, @CheckForNull Throwable t) {
 
     @Override public void deleteArtifacts() throws IOException {
         super.deleteArtifacts();
-        StashManager.clearAll(this);
+        try {
+            StashManager.clearAll(this, /* core API defines no log sink for this */ new LogTaskListener(LOGGER, Level.FINE));
```

Comment:
…since `BuildDiscarder` is called from a new build and then deletes artifacts from older builds. The `logRotate` call is currently also asynchronous and done after closing the logger for the new build in `finish`. Both things _could_ be changed, but we would still need a core API change to allow us to thread the right `TaskListener` through `Job.logRotate` → `BuildDiscarder.perform` (in `LogRotator`) → `Run.deleteArtifacts`. Also through `LogRotator.perform` → `Run.delete` → `Run.deleteArtifacts` as of https://github.com/jenkinsci/jenkins/pull/3484 (which is in 2.121.2). A lot of work just to display

> Deleted … stash(es) from …

Currently artifact deletion also not have any available `TaskListener`, though that is not touched by this PR.